src/WebJobs.Script/Diagnostics/OpenTelemetry/OpenTelemetryConfigurationExtensions.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Diagnostics.Tracing;
+using Azure.Monitor.OpenTelemetry.Exporter;
+using Azure.Monitor.OpenTelemetry.LiveMetrics;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using OpenTelemetry.Logs;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Resources;
+using OpenTelemetry.Trace;
+
+namespace Microsoft.Azure.WebJobs.Script.Diagnostics.OpenTelemetry
+{
+    internal static class OpenTelemetryConfigurationExtensions
+    {
+        internal static void ConfigureOpenTelemetry(this ILoggingBuilder loggingBuilder, HostBuilderContext context)
+        {
+            string azMonConnectionString = GetConfigurationValue(EnvironmentSettingNames.AppInsightsConnectionString, context.Configuration);
+            bool enableOtlp = false;
+            if (!string.IsNullOrEmpty(GetConfigurationValue(EnvironmentSettingNames.OtlpEndpoint, context.Configuration)))
+            {
+                enableOtlp = true;
+            }
+
+            loggingBuilder
+                .AddOpenTelemetry(o =>
+                {
+                    o.SetResourceBuilder(ConfigureResource(ResourceBuilder.CreateDefault()));
+                    if (enableOtlp)
+                    {
+                        o.AddOtlpExporter();
+                    }
+                    if (!string.IsNullOrEmpty(azMonConnectionString))
+                    {
+                        o.AddAzureMonitorLogExporter(options => options.ConnectionString = azMonConnectionString);
+                    }
+                    o.IncludeFormattedMessage = true;
+                    o.IncludeScopes = false;
+                })
+                // These are messages piped back to the host from the worker - we don't handle these anymore if the worker has OpenTelemetry enabled.
+                // Instead, we expect the user's own code to be logging these where they want them to go.
+                .AddFilter<OpenTelemetryLoggerProvider>("Host.*", _ => !ScriptHost.WorkerOpenTelemetryEnabled)
```

Comment:
We should verify this, I don't think it is blanket true statement. There may be code in the host which still starts with these category names. `Azure.*` for sure, as the host will use Azure SDKs. `Microsoft.Azure.WebJobs.*` also - those assemblies are all in the host, not the worker.

`Function.*` and `Host.*` we will need to see.

Also, if we are already sending a capability from the worker back to the host, can the worker just opt to _not_ send telemetry back to the host when it emits that capability? We already do this for app insights in dotnet isolated.

Reply:
Removed `Host.*` and `Microsoft.Azure.WebJobs.*`. There's lot of noise coming from `Azure.*`. I will evaluate this again.

We still have to support FIleLogging and DiagnosticsSettings, we need worker logs in the host for this.